### DISCUSSION TITLE
solution.get_TabularData() causes an error

### DISCUSSION THREAD

**Discussion by anthony-walker** [ansys/pymechanical#873] [Q&A] [answered]
Calling `solution.get_TabularData()` in the context of `App.Model.Analyses[0].Solution` causes the following error on Windows. Has anyone encountered this and a way to fix it?

```
 System.Runtime.InteropServices.COMException:  (0x80004005): Error HRESULT E_FAIL has been returned from a call to a COM component.
   at Ansys.Common.Interop.DSGfx._DSCompositeDataProviderClass.get_Table(UInt32 uId)
   at Ansys.ACT.Automation.Mechanical.Solution.get_TabularData()
```

**Comment by koubaa** [comment, marked as answer]
@anthony-walker This is related to Mechanical Scripting, which is documented [here](https://mechanical.docs.pyansys.com/version/stable/user_guide_scripting/index.html). The API reference can be reached from the ACT API reference guide, linked there under "Additional Resources"[1]. There, it can be seen that the `TabularData`[2] property is not mentioned as an API of the `Solution` object. In our terminology, this means that it is not _Published_. 

The bottomline is that just because something is reachable with Python it doesn't mean that it is `Published`. When doing python scripting inside of the Mechnical user interface, only the `Published` APIs will appear when using autocomplete. I have to admit that this isn't ideal for the PyMechanical user, since there's no easy way for PyMechanical to tell whether a given property or method is `Published`.

----

1. In the future, we plan to directly host the Mechanical API reference somewhere more convenient to access.
2. `get_TabularData()` appears to be an alias of the `TabularData` property. This looks like an implementation detail of how python<->C# interop works for properties. For properties, using the direct accessor is preferred (and indeed only the direct accessor is documented in our API reference guide). i.e. `App.Model.Name` is preferred over `App.Model.get_Name()`
